### ISSUE TITLE
1.0.0 Release Trim Error

### ISSUE THREAD

**Issue by 4ndygu** [Dashlane/dashlane-cli#42]
Hi! I'm getting this error after submitting my user/pass/code on the 1.0.0 dashlane cli:

```
TypeError: Cannot read properties of undefined (reading 'trim')
    at PrintableVaultCredential.toString (/snapshot/dashlane-cli/dist/types.js:9:44)
    at /snapshot/dashlane-cli/dist/middleware/getPasswords.js:84:46
    at Array.map (<anonymous>)
    at selectCredential (/snapshot/dashlane-cli/dist/middleware/getPasswords.js:82:41)
    at async getPassword (/snapshot/dashlane-cli/dist/middleware/getPasswords.js:92:32)
    at async Command.<anonymous> (/snapshot/dashlane-cli/dist/index.js:55:9)
    at async Command.parseAsync (/snapshot/dashlane-cli/node_modules/commander/lib/command.js:917:5)
```

**Comment by Mikescops** [comment]
Hello,

Are you using node 16?

**Comment by 4ndygu** [comment]
Hi! I downloaded the binary directly, and my `node -v` returns 18.2.0.

**Comment by Mikescops** [comment]
Can you share your OS info?

@jboillot fyi

**Comment by jboillot** [comment]
Hello!
Thank you for submitting this interesting issue.
I currently cannot reproduce or understand what is causing this bug.
I suppose it is related to one of your passwords whose 'email' and 'login' fields are empty.
So, if you want to access your vault 
I will continue to investigate but, if you want a temporary fix I would recommend using jq to retrieve your passwords with neither email nor login: `dcli p --output json | jq '.[] | select(.email | not) | select(.login | not) | .title'`.
After modifying those entries you would have to run `dcli s` again.

**Comment by 4ndygu** [comment]
Thanks! It seems like the json output solution works for me, though the raw does not. Here's the output with `--debug` on, though admittedly less helpful:

```
debug:   Connected to database.
debug:   Retrieving:
TypeError: Cannot read properties of undefined (reading 'trim')
    at PrintableVaultCredential.toString (/snapshot/dashlane-cli/dist/types.js:9:44)
    at /snapshot/dashlane-cli/dist/middleware/getPasswords.js:84:46
    at Array.map (<anonymous>)
    at selectCredential (/snapshot/dashlane-cli/dist/middleware/getPasswords.js:82:41)
    at async getPassword (/snapshot/dashlane-cli/dist/middleware/getPasswords.js:92:32)
    at async Command.<anonymous> (/snapshot/dashlane-cli/dist/index.js:55:9)
    at async Command.parseAsync (/snapshot/dashlane-cli/node_modules/commander/lib/command.js:917:5)
```

**Comment by jboillot** [comment]
Yes this is because the interactive filtering is broken, if you already know which password you're looking for you can directly use `dcli p github.com` to avoid this bug.

**Comment by 4ndygu** [comment]
Thank you! Honestly - this product is a lifesaver. I'll hopefully get some time to look at this at some point too!

**Comment by 4ndygu** [comment]
Oh i see, I think I messed up and just assumed that the output of `dcli p` was to drop all passwords. This error is just a misparse of an undefined filter. I'll close this - my bad lol.

**Comment by jboillot** [comment]
No you were right there is indeed a bug and `dcli p` should show you all possible passwords. I'll keep investigating about that :)

**Comment by Mikescops** [comment]
Should be fixed in `v1.1.0`, please re-open if there is still an issue.
